apis/apiextensions/fn/proto/v1beta1/run_function.proto

```diff
@@ -90,6 +90,9 @@ message State {
 
   // The state of any composed resources.
   map<string, Resource> resources = 2;
+
+  // The environment passed to the Function pipeline.
```

Comment:
Should we add some kind of `THIS IS AN ALPHA FIELD` warning like we do on our CRDs? Something that communicates that this field is more at-risk of changing than the rest of the API.

Reply:
As to how it might change - if we did make `environment` an array or map of resources I could imagine reflecting that here. So something more like:

```proto
map<string, google.protobuf.Struct> environment = 3;
```

Reply:
Yes, you are right. I will update it 👍

Reply:
Added the alpha warning as suggested 👍 

About the type of the environment field, wouldn't a `google.protobuf.Struct` already accomodate such a change? currently the environment is fully unstructured:
```go
// Environment defines unstructured data.
type Environment struct {
	unstructured.Unstructured
}
```
But if we decided to have two separate sections, "data" and "resources" as I'm [suggesting](https://github.com/crossplane/crossplane/issues/4583#issuecomment-1719193018), we would just have to populate the Environment in a different way, but we shouldn't need to change the code here.

Reply:
I agree we wouldn't _have to_, in that a `google.protobuf.Struct` can pretty much model anything. It does however seem to me that if we know the environment will always have a certain shape (i.e. data and array of resources) it would be a better experience for folks building functions for us to reflect that shape in our 'schema' as much as possible.